fieldservice_sale_crm_lead_schduration/models/sale_order.py

```diff
@@ -0,0 +1,13 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _prepare_fsm_values(self, **kwargs):
+        prepare = super(SaleOrder, self)._prepare_fsm_values(**kwargs)
+        if self.opportunity_id.duration:
+            prepare["scheduled_duration"] = self.opportunity_id.duration 
```

Comment:
A security issue could be fired here: user is a low Sales profile, and CRM opportunity belongs to another user than Sales order. In that case, a Record Rule error is raised. Then, better we'll do this:
```suggestion
            prepare["scheduled_duration"] = self.sudo().opportunity_id.duration 
```